### PR TITLE
refactor: map provider and splitter works on buffer instead of stream

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/retiler/interfaces.ts [hunk 1]
```diff
@@ -13,11 +12,11 @@ export interface JobQueueProvider {
 }
 
 export interface MapProvider {
-  getMapStream: (bbox: BoundingBox, mapWidth: number, mapHeight: number) => Promise<Readable>;
+  getMapBuffer: (bbox: BoundingBox, mapWidth: number, mapHeight: number) => Promise<Buffer>;
```
Comment (CptSchnitz) [inline src/retiler/interfaces.ts:15]: change to get map

File: src/retiler/mapSplitterProvider/sharp.ts [hunk 2]
```diff
@@ -1,36 +1,34 @@
...
 import { TileWithBuffer } from '../types';
+import { bufferToStream } from '../../common/utils';
 
-// see https://sharp.pixelplumbing.com/api-utility for more global sharp properties that can affect performance (concurrency, cache)
-// add implementation to these properties if needed
 export class SharpMapSplitter implements MapSplitterProvider {
-  public async splitMap(tile: Tile, stream: Readable): Promise<TileWithBuffer[]> {
-    const metatile = tile.metatile ?? 1;
+  public async splitMap(tile: Tile, buffer: Buffer): Promise<TileWithBuffer[]> {
+    const stream = bufferToStream(buffer);
```
Comment (CptSchnitz) [inline src/retiler/mapSplitterProvider/sharp.ts:10]: ?

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/retiler/mapSplitterProvider/sharp.ts (hunk 2 above)
Comment (melancholiai) [reply src/retiler/mapSplitterProvider/sharp.ts]: done.

File: src/retiler/interfaces.ts (hunk 1 above)
Comment (melancholiai) [reply src/retiler/interfaces.ts]: done.

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**